src/combine-results.ts

```diff
@@ -94,34 +94,46 @@ async function run(): Promise<void> {
       title,
     });
 
-    const csvs: string[] = [];
-    const resultsMd = await Promise.all(
+    const csvs: string[] = await Promise.all(
       resultArtifacts.map(async function (response) {
         const csv = path.join(response.downloadPath, "results.csv");
         const csvDest = path.join("results", response.artifactName);
         await mv(csv, csvDest);
-        csvs.push(csvDest);
-
+        return csvDest;
+      })
+    );
+    const resultsMd = await Promise.all(
+      resultArtifacts.map(async function (response) {
         const repoName = fs.readFileSync(
           path.join(response.downloadPath, "nwo.txt"),
           "utf-8"
         );
-        const output = await getExecOutput("wc", ["-l", csvDest]); // TODO: preferably we would do this during results interpretation
-        const results = parseInt(output.stdout.trim(), 10) - 1;
+        const resultCount = parseInt(
+          fs.readFileSync(
+            path.join(response.downloadPath, "resultcount.txt"),
+            "utf-8"
+          ),
+          10
+        );
 
-        if (results > 0) {
+        if (resultCount > 0) {
           const md = path.join(response.downloadPath, "results.md");
           const comment = await octokit.rest.issues.createComment({
             owner: context.repo.owner,
             repo: context.repo.repo,
             issue_number: issue.data.number,
             body: fs.readFileSync(md, "utf8"),
           });
-          return `| ${repoName} | [${results} result(s)](${comment.data.html_url}) |`;
+          return `| ${repoName} | [${resultCount} result(s)](${comment.data.html_url}) |`;
         }
         return `| ${repoName} | _No results_ |`;
       })
     );
+    const resultsIndex = createResultIndex(resultArtifacts);
```

Comment:
```suggestion
    const resultsIndex = await createResultIndex(resultArtifacts);
```

Perhaps `JSON.stringify` waits internally somehow, but this feels more "correct"

Reply:
No, I think we were stringifying a promise here, so we do need to `await`. Another thing you could do is explicitly add a type to `resultsIndex` to assert it is what we want it to be, e.g.
```suggestion
    const resultsIndex: ResultIndexItem[] = await createResultIndex(resultArtifacts);
```

Reply:
Thank you, that makes sense!

Reply:
Ah! I think the integration test "caught" this (the results-index/index.json at https://github.com/dsp-testing/qc-run2/actions/runs/1449152312 is just `{}`). That integration test just checks that we aren't crashing though (which is fine).